### PR TITLE
Fix: sync project from app crd to velaux

### LINKED ISSUE #5352: [Bug] Specify the env of a project to take over the application, but the project is the default project
**Describe the bug**

Specify the env of a project to take over the application, but the project is the default project rather than the project to which the env belongs.
image
image
image

**To Reproduce**

**Expected behavior**

**Screenshots**

**KubeVela Version**
1.7.0

**Cluster information**

**Additional context**

### PR DESCRIPTION
**Pull request by nuclearwu**
Signed-off-by: wuzhongjian <wuzhongjian_yewu@cmss.chinamobile.com>

### Description of your changes
Specify the env of a project to take over the application, the app project should be the project to which env belongs.

Fixes #5352 

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Comment by nuclearwu** [comment]
@barnettZQG

File: pkg/apiserver/event/sync/convert.go [hunk 1]
```diff
@@ -190,7 +190,7 @@ func (c *CR2UX) generateEnv(ctx context.Context, defaultProject string, envNames
 				env.Targets = append(env.Targets, name)
 			}
 		}
-		return env, "", nil
+		return env, env.Project, nil
```
Comment (barnettZQG) [inline pkg/apiserver/event/sync/convert.go:193]: Nice, please add a unit test to cover this case.

Comment (nuclearwu) [reply pkg/apiserver/event/sync/convert.go:193]: done

**Review by barnettZQG — APPROVED** [review]
LGTM

**Comment by nuclearwu** [comment]
@barnettZQG It is necessary to add the backport label to fix the historical version.

**Pull request merged**